analyzers/src/SonarAnalyzer.CSharp/Rules/NewGuidShouldNotBeUsed.cs

```diff
@@ -25,35 +25,63 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 using SonarAnalyzer.Helpers;
+using SonarAnalyzer.Wrappers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
     public sealed class NewGuidShouldNotBeUsed : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S4581";
-        private const string MessageFormat = "Use 'Guid.NewGuid()' or 'Guid.Empty' or add arguments to this Guid instantiation.";
+        private const string DiagnosticId = "S4581";
+        private const string MessageFormat = "Use 'Guid.NewGuid()', 'Guid.Empty' or the constructor with arguments.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         protected override void Initialize(SonarAnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            DetectIssueInConstructors(context);
+            DetectIssueInDefaultExpressions(context);
+        }
+
+        private static void DetectIssueInConstructors(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var objectCreationSyntax = ObjectCreationFactory.Create(c.Node);
+                if (objectCreationSyntax.ArgumentList?.Arguments.Count == 0
+                    && objectCreationSyntax.MethodSymbol(c.SemanticModel) is IMethodSymbol methodSymbol
```

Comment:
Currently `is IMethodSymbol` is only doing a null check since `MethodSymbol` returns already a `IMethodSymbol`. This is equivalent to `is { }`.
```suggestion
                    && objectCreationSyntax.MethodSymbol(c.SemanticModel) is { } methodSymbol
```